AiTableTopGameMaster.Core/Services/OutputReviewAgent.cs

```diff
@@ -0,0 +1,130 @@
+using Microsoft.Extensions.Logging;
+using Microsoft.SemanticKernel;
+using Microsoft.SemanticKernel.Agents;
+using Microsoft.SemanticKernel.ChatCompletion;
+
+namespace AiTableTopGameMaster.Core.Services;
+
+/// <summary>
+/// Agent-based implementation of output reviewer that uses AI to detect undesirable behaviors
+/// in game master output.
+/// </summary>
+public class OutputReviewAgent : IOutputReviewer
```

Comment:
I like us using an Agent here for the review agent, but I wonder if we're trying to do the wrong thing: instead of parsing its output, we could be providing plugins to the agent for acceptable and unacceptable and have them both take in reasonings. We could then require the agent to call one of these two plugins and make sure these are the only two plugins it sees.